docs/query-metadata-style-guide.md

```diff
@@ -0,0 +1,152 @@
+*   [Python queries](https://wiki.semmle.com/display/PYTHON/Python+queries)
+
+## Metadata area
+
+Query file metadata contains important information which defines the name and purpose of the query. In order to help others use your query, and to ensure that the query works correctly on LGTM.com, you should include all of the required information outlined below in the metadata, and as much of the optional information as as possible. For further information on query metadata see [Query file requirements](https://help.semmle.com/wiki/display/SD/Query+file+requirements).
+
+### Query name `@name`
+
+You must specify an `@name` property for your query. This property defines the display name for the query. Query names should use sentence capitalization, but not include a full stop. Filter queries should specify the results that are excluded when you run the query. See the following examples for more detail:
+
+*   `@name Access to variable in enclosing class`
+*   `@name Array argument size mismatch`
+*   `@name Reference equality test on strings`
+*   `@name Return statement outside function`
+
+
+### Query descriptions `@description`
+
+You must define an `@description` property for your query. This property defines a short help message. Query descriptions should be written as a sentence or short-paragraph of plain prose, with sentence capitalization and full stop. The preferred pattern for alert queries is  `Syntax X causes behavior Y.` Any code elements included in the description should be enclosed in single quotes. For example:
+
+
+*   `@description Using a format string with an incorrect format causes a 'System.FormatException'.`
+*   `@description Commented-out code makes the remaining code more difficult to read.`
+
+### Query ID `@id`
+
+You must specify an `@id` property for your query. It must be unique in the Semmle namespace and should follow the standard Semmle convention. That is, it should begin with the 'language code' for the language that the query analyzes followed by a forward slash. The following language codes are supported:
+
+*   C and C++: `cpp`
+*   C#: `cs`
+*   COBOL: `cobol`
+*   Java: `java`
+*   JavaScript and TypeScript: `js`
+*   Python: `py`
+
+The `@id` should consist of a short noun phrase that identifies the issue that the query highlights. For example:
+
+
+
+*   `@id cs/command-line-injection`
+*   `@id java/string-concatenation-in-loop`
+
+Note, `@id` properties should be consistent for queries that highlight the same issue for different languages. For example, the following queries identify format strings which contain unsanitized input in Java and C++ code respectively:
+
+
+
+*   `@id java/tainted-format-string`
+*   `@id cpp/tainted-format-string`
+
+
+### Query type `@kind`
+
+`@kind` is a required property that defines the type of query. The main query types are:
+
+
+
+*   alerts (`@kind problem`)
+*   alerts containing path information (`@kind path-problem`)
```

Comment:
I don't think this information about `path-problem` is covered on our internal wiki.  Perhaps we should add it or link to this doc page from https://wiki.semmle.com/display/IN/Writing+queries+and+query+help?

More generally, I think there's a widespread issue (not specific to this PR) of duplicating information between our internal wiki and publically available documentation, with possible consequences including inconsistencies, and Semmle developers with access to both forgetting to read / update one or other source of information.  Perhaps in these cases we should aim for the wiki page to be a small stub page that only (1) provides a prominent link to the doc page and (2) provides any additional information and advice that for whatever reason we don't deem suitable for the general audience.

Reply:
Fair point. I'll be adding a 'query help style-guide' in a future PR, so a good solution for that wiki page would be to link to these docs.